presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -346,7 +347,15 @@ public static ConnectorPageSource createParquetPageSource(
                         List<String> nestedColumnPath = nestedColumnPath(pushedDownSubfield);
                         Optional<ColumnIO> columnIO = findNestedColumnIO(lookupColumnByName(messageColumnIO, pushedDownSubfield.getRootName()), nestedColumnPath);
                         if (columnIO.isPresent()) {
-                            fieldsBuilder.add(constructField(type, columnIO.get()));
+                            if (isPushedDownSubfield(column)) {
```

Comment:
In line 258, we returned 'AggregatedParquetPageSource' if it is aggregated. Does this fix also apply to Aggregated type?